### ISSUE TITLE
Constant template should not be flagged by linter

### ISSUE THREAD

**Issue by alex-vinnik-sp** [grafana/dashboard-linter#99]
I have a constant variable in a dashboard and it looks like it gets flagged by linter 
```
     {
        "hide": 2,¬
        "name": "instance",¬
        "query": ".*",¬
        "skipUrlSync": false,¬
        "type": "constant"¬
      }
```
```
$HOME/go/bin/dashboard-linter lint mydashboard.json
```
```
[❌] Dashboard 'My dashboard' instance template should use datasource '$datasource', is currently ''
```
Should it be?

**Comment by rgeyer** [comment]
Only two years later... Sorry.

This is triggering, not because it's a constant, but because it's named "instance". The dashboard-linter has very opinionated rules about having an instance templated variable, and how it's used.

I suppose we *could* ignore "constant" template variables, but the expected behavior for an instance variable would be that it would be queried from the underlying data source, and thus, should have a data source defined, and thus should have a templated data source.
